src/scripts/sync_synonym.py

```diff
@@ -0,0 +1,184 @@
+"""Create outputs for syncing synonyms between Mondo and its sources."""
+import os
+import sys
+from argparse import ArgumentParser
+from pathlib import Path
+from typing import Union, List, Tuple, Set, Dict
+
+import pandas as pd
+from oaklib import get_adapter
+from oaklib.implementations import SqlImplementation
+from oaklib.types import CURIE, PRED_CURIE
+
+HERE = Path(os.path.abspath(os.path.dirname(__file__)))
+SRC_DIR = HERE.parent
+PROJECT_ROOT = SRC_DIR.parent
+sys.path.insert(0, str(PROJECT_ROOT))
+from src.scripts.utils import PREFIX_MAP, get_owned_prefix_map
+
+
+def _query_synonyms(ids: List[CURIE], db: SqlImplementation) -> pd.DataFrame:
+    """Get synonym triples from sqlite DB"""
+    tups: Set[Tuple[CURIE, PRED_CURIE, str]] = set()
+    for _id in ids:
+        for pred, alias in sorted(db.alias_relationships(_id)):
+            # - filter all but synonyms: prior in pipeline have already converted other props such as 'label' to syns
+            # todo: is this the best filter? is there more that can appear that should be filtered out? Should instead
+            #  not filter by things we want out, but by the list of synonym preds we want in?
+            if pred == 'rdfs:label':
+                continue
+            tups.add((_id, pred, alias))
+    df = pd.DataFrame(tups, columns=['curie', 'synonym_scope', 'synonym'])
+    return df
+
+
+def _filter_non_intersecting_rows(base_df: pd.DataFrame, intersecting_df: pd.DataFrame, join_on: List[str]) -> pd.DataFrame:
+    """Filter out all rows are not in common, given a `join_on`. """
+    merge_df = base_df.merge(intersecting_df, on=join_on, how='left', indicator=True)
+    base_df2 = merge_df[merge_df['_merge'] == 'left_only'].drop(columns=['_merge']).rename(
+        columns=lambda x: x[:-2] if x.endswith('_x') else x)[list(base_df.columns)]
+    return base_df2
+
+
+# TODO
+def sync_synonyms(
+    ontology_db_path: Union[Path, str], mondo_synonyms_path: Union[Path, str],
+    mondo_mappings_path: Union[Path, str], onto_config_path: Union[Path, str], outpath_added: Union[Path, str],
+    outpath_confirmed: Union[Path, str], outpath_deleted: Union[Path, str], outpath_updated: Union[Path, str]
+):
+    """Create outputs for syncing synonyms between Mondo and its sources."""
+    # Iterate over mappings
+    mappings_df = pd.read_csv(mondo_mappings_path, sep='\t', comment='#')
+    mappings_df = mappings_df[mappings_df['predicate_id'] == 'skos:exactMatch']
```

Comment:
## Only iter over exact matches
@matentzn I'm doing this additional filter as a just-in-case. Currently, `mondo.sssom.tsv` only has exact matches. Was this always the case? I suppose that, given our typical naming conventions it ought to be called `mondo-exact.sssom.tsv`.

In any case, this line just ensures that, should non-exact matches make their way in, we filter them out.

This is hypothetical, but: should that ever change and we would have non-exact matches, do we indeed want to filter them out? For example, shouldn't it be theoretically possible to have a `narrowMatch`, and to import one of its synonyms as a `narrowSynonym`?

Reply:
@matentzn You ended up answering this question in [another comment](https://github.com/monarch-initiative/mondo-ingest/pull/93#discussion_r1680513122):
> We should only ever sync content across "exact" matches

That clarifies my main question I wanted to double check.

Regarding why we don't import stuff for broad/narrow/related, my guess is that it is simply too much of a curation burden.

@twhetzel I'm marking this one resolved.